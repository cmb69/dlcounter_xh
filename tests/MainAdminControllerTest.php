<?php

namespace Dlcounter;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\Jquery;
use Plib\View;

class MainAdminControllerTest extends TestCase
{
    public function testRendersStats(): void
    {
        $dbService = $this->createStub(DbService::class);
        $dbService->method("readDb")->willReturn($this->downloads());
        $sut = new MainAdminController(
            "../plugins/",
            $dbService,
            $this->createStub(Jquery::class),
            $this->view()
        );
        Approvals::verifyHtml($sut->defaultAction(new FakeRequest()));
    }

    private function downloads(): array
    {
        return [
            (object) ["name" => "foo.txt", "time" => 12345678],
            (object) ["name" => "bar.txt", "time" => 23456789],
        ];
    }

    private function view(): View
    {
        return new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["dlcounter"]);
    }
}
