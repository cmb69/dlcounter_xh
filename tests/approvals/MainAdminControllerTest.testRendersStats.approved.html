
<h1>Dlcounter â€“ Statistics</h1>
<div id="dlcounter_stats">
  <h4>Totals</h4>
  <table id="dlcounter_summary_table" class="tablesorter">
    <thead>
      <tr>
        <th>File</th>
        <th>Count</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>foo.txt</td>
        <td>1</td>
      </tr>
      <tr>
        <td>bar.txt</td>
        <td>1</td>
      </tr>
    </tbody>
  </table>
  <h4>Individual Downloads</h4>
  <table id="dlcounter_details_table" class="tablesorter">
    <thead>
      <tr>
        <th>Date</th>
        <th>File</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>1970-05-23 21:21:18</td>
        <td>foo.txt</td>
      </tr>
      <tr>
        <td>1970-09-29 11:46:29</td>
        <td>bar.txt</td>
      </tr>
    </tbody>
  </table>
</div>
<script>
jQuery(function($) {
    $("#dlcounter_stats h4").click(function () {
        $(this).next().toggle();
    });
    $("#dlcounter_stats .tablesorter").tablesorter();
});
</script>
